src/tbp/monty/frameworks/config_utils/make_dataset_configs.py

```diff
@@ -112,7 +112,7 @@ class EnvInitArgsMontyWorldMultiObjectScenes:
 @dataclass
 class OmniglotDatasetArgs:
     env_init_func: Callable = field(default=OmniglotEnvironment)
-    env_init_args: Dict = field(default_factory=lambda: dict())
+    env_init_args: Dict = field(default_factory=lambda: {})
```

Comment:
Note to self: the change here really should be to remove the `lambda` and use the collection function directly.

Reply:
Actually, I take it back, we can't do that, because the factory has to create a **new** dictionary each time. I'm pretty sure it's the same gotcha that happens if you put a list or dictionary literal as the default argument to a function or method. It gets shared between all calls of the function as a single object.